src/sagemaker/jumpstart/types.py

```diff
@@ -1091,31 +1086,38 @@ def to_json(self) -> Dict[str, Any]:
         json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
         return json_obj
 
-    def resolve(self) -> JumpStartPresetComponent:
-        """Returns the final config that is resolved from the list of components."""
+    @property
```

Comment:
does `@property` ensure this logic is only performed once and then cached?

Reply:
I don't think it [gets cached with this tag automatically](https://stackoverflow.com/questions/17330160/how-does-the-property-decorator-work-in-python), that's why I stored it in a variable  `self.resolved_preset_config`